src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java

```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
+        when(asylumCase.read(FTPA_RESPONDENT_SUBMITTED, String.class)).thenReturn(Optional.of("YES"));
+
+        when(asylumCase.read(FTPA_RESPONDENT_DECISION_OUTCOME_TYPE, String.class)).thenReturn(Optional.of("granted"));
+
+
+        PreSubmitCallbackResponse<AsylumCase> callbackResponse =
+            ftpaDecisionMidEvent.handle(MID_EVENT, callback);
+
+        assertNotNull(callback);
+        assertEquals(asylumCase, callbackResponse.getData());
+        final Set<String> errors = callbackResponse.getErrors();
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_RESPONDENT_DECISION_REASONS_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_ftpa_respondent_decision_reasons_notes_visibility_when_partially_granted() {
+
+        when(callback.getEvent()).thenReturn(Event.LEADERSHIP_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("respondent"));
+        when(asylumCase.read(FTPA_RESPONDENT_SUBMITTED, String.class)).thenReturn(Optional.of("YES"));
+
+        when(asylumCase.read(FTPA_RESPONDENT_DECISION_OUTCOME_TYPE, String.class)).thenReturn(Optional.of("partiallyGranted"));
+
+
+        PreSubmitCallbackResponse<AsylumCase> callbackResponse =
+            ftpaDecisionMidEvent.handle(MID_EVENT, callback);
+
+        assertNotNull(callback);
+        assertEquals(asylumCase, callbackResponse.getData());
+        final Set<String> errors = callbackResponse.getErrors();
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_RESPONDENT_DECISION_REASONS_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_appellant_notice_of_decision_set_aside_visibility_when_granted() {
+
+        when(callback.getEvent()).thenReturn(Event.RESIDENT_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("appellant"));
+        when(asylumCase.read(FTPA_APPELLANT_SUBMITTED, String.class)).thenReturn(Optional.of("YES"));
+
+        when(asylumCase.read(FTPA_APPELLANT_RJ_DECISION_OUTCOME_TYPE, String.class)).thenReturn(Optional.of("granted"));
+
+
+        PreSubmitCallbackResponse<AsylumCase> callbackResponse =
+            ftpaDecisionMidEvent.handle(MID_EVENT, callback);
+
+        assertNotNull(callback);
+        assertEquals(asylumCase, callbackResponse.getData());
+        final Set<String> errors = callbackResponse.getErrors();
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_APPELLANT_NOTICE_OF_DECISION_SET_ASIDE_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_appellant_notice_of_decision_set_aside_visibility_when_partially_granted() {
+
+        when(callback.getEvent()).thenReturn(Event.RESIDENT_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("appellant"));
```

Comment:
"appellant" - Shouldn't this be a constant taken from FTPA_APPLICANT_TYPE ?

Reply:
It is again defined as String type and there is no constant defined.